### PR TITLE
Api service layer unit tests

### PR DESCRIPTION
**Pull request by NikPeters-Atos**
Added a Repository layer which abstracts the database context away from the service layer, thereby allowing the Service layer to be unit tested much easier. This therefore makes a more sensible approach to the data layer. The repository is merely a wrapper around the DbContext with no business logic, thereby requiring no unit tests itself.

### CODE REVIEW

File: Portal/Services/WasteService.cs [hunk 1]
```diff
@@ -25,7 +25,8 @@ public async Task<DuringWhichMonthRequestViewModel> GetCurrentQuarter(int journe
             var duringWhichMonthRequestViewModel = new DuringWhichMonthRequestViewModel
             {
                 JourneyId = journeyId,
-                WasteType = await _httpWasteService.GetWasteType(journeyId)
+                // We're not part of a journey yet, so this can't really be hooked up
```
Comment (shehzad-ismail8) [inline Portal/Services/WasteService.cs:28]: Does this mean that we're going to have to hard code the waste type in our views for the time being?

File: Waste.API/Services/WasteService.cs [hunk 2]
```diff
@@ -28,68 +27,76 @@ public async Task<int> CreateJourney()
                 CreatedBy = "DEVELOPER"
             };
 
-            _wasteContext.WasteJourney.Add(journeyRecord);
-            await _wasteContext.SaveChangesAsync();
+            await _wasteRepository.Add(journeyRecord);
+
             return journeyRecord.Id;
```
Comment (shehzad-ismail8) [inline Waste.API/Services/WasteService.cs:32]: Do we need to return journeyRecord.Id if an id is being returned in the sealed Repository class?

**Review by shehzad-ismail8 — COMMENTED** [review]
Looks good to me, a couple of comments.

File: Portal/Services/WasteService.cs (hunk 1 above)
Comment (NikPeters-Atos) [reply Portal/Services/WasteService.cs:28]: Or not have it at all, yes. For example, does your page work with this code and an empty database... (clue: it won't) There's really no point in having this bit at the moment

File: Waste.API/Services/WasteService.cs (hunk 2 above)
Comment (NikPeters-Atos) [reply Waste.API/Services/WasteService.cs:32]: The id is only generated once it has been saved to the database: it's an auto incrementing field, and therefore we only know about it after we have succesfully saved to the database.

**Review by prasadcherukuri — APPROVED** [review]

**Pull request merged**